src/Microsoft.AspNetCore.SignalR/Proxies.cs

```diff
@@ -54,12 +55,28 @@ public Task InvokeAsync(string method, params object[] args)
         }
     }
 
+    public class AllClientsExceptProxy<THub> : IClientProxy
+    {
+        private readonly HubLifetimeManager<THub> _lifetimeManager;
+        private readonly List<string> _excludedIds;
+
+        public AllClientsExceptProxy(HubLifetimeManager<THub> lifetimeManager, List<string> excludedIds)
```

Comment:
We never need to write to the collection, so we should avoid using a writable interface, hence `IReadOnlyCollection` or `IReadOnlyList` would be a better option.

In general, we want to lead users towards good performance without them having to know how our code is implemented, so we try to avoid leading users to a path where they have to consider the performance of collection objects they give us. It's our job to make processing the collection fast, not the user's.